### PR TITLE
macros: avoid use of Url::join where necessary

### PR DESCRIPTION
**Pull request by mathstuf**

### CODE REVIEW

File: src/macros.rs [hunk 1]
```diff
@@ -21,7 +21,11 @@ macro_rules! from {
                 if f.request.is_ok() {
                     // We've checked that this works
                     let mut req = f.request.unwrap();
-                    let url = req.url().join($e)
+                    // We don't use `url.join` here because it uses a trailing slash as an
+                    // indicator, but some endpoints are both an "file" and "directory", so we
```
Comment (mgattozzi) [inline src/macros.rs]: %s/both an/both a

File: src/macros.rs [hunk 2]
```diff
@@ -237,7 +241,11 @@ macro_rules! func_client{
             if self.request.is_ok() {
                 // We've checked that this works
                 let mut req = self.request.unwrap();
-                let url = req.url().join($e)
+                // We don't use `url.join` here because it uses a trailing slash as an
+                // indicator, but some endpoints are both an "file" and "directory", so we
```
Comment (mgattozzi) [inline src/macros.rs]: Same as above

File: src/macros.rs (hunk 1 above)
Comment (mathstuf) [reply src/macros.rs]: Fixed.

**Comment by mgattozzi** [comment]
Ran the tests locally looks like the URL got messed up again:

URL: Ok("https://api.github.com//repos")

**Comment by mathstuf** [comment]
Updated again :) . The top-level can use `.join` since there's already a trailing slash.

**Comment by mgattozzi** [comment]
I'm still getting the URL as https://api.github.com//repos when I run the tests

**Comment by mgattozzi** [comment]
@mathstuf I'm going to close this until all the stuff with reqwest can get figured out again. Feel free to open this up again when that gets worked through.

**Pull request closed without merging**